src/components/withIntl.js

```diff
@@ -8,62 +9,71 @@ function getDisplayName(Component) {
 }
 
 const IntlContext = createContext(null);
-const {
-  Consumer: IntlConsumer,
-  Provider: IntlProvider
-} = IntlContext
+const {Consumer: IntlConsumer, Provider: IntlProvider} = IntlContext;
 
-export const Provider = IntlProvider
-export const Context = IntlContext
+export const Provider = IntlProvider;
+export const Context = IntlContext;
 
-export default function withIntl(WrappedComponent, options = {}) {
+export default function withIntl(componentOrOptions, options) {
+  if (isValidElementType(componentOrOptions)) {
+    return createWrapper(componentOrOptions, options);
```

Comment:
What do you think about instead using:
```js
return createWrapper.call({ options }, componentOrOptions);
```
This would allow `createWrapper()` to always read the options from `this`.

Reply:
Sounds good, I'll change it